runtime/lua/vim/lsp/buf.lua

```diff
@@ -280,14 +280,14 @@ function M.rename(new_name)
     end
     if result and result.placeholder then
       opts.default = result.placeholder
-      if not new_name then npcall(vim.ui.input, opts, on_confirm) end
+      if not new_name then npcall(vim.ui.input, opts, request_rename) end
     elseif result and result.start and result['end'] and
       result.start.line == result['end'].line then
       local line = vfn.getline(result.start.line+1)
       local start_char = result.start.character+1
       local end_char = result['end'].character
       opts.default = string.sub(line, start_char, end_char)
-      if not new_name then npcall(vim.ui.input, opts, on_confirm) end
+      if not new_name then npcall(vim.ui.input, opts, request_rename) end
     else
```

Comment:
This isn't quite the right behavior btw. The spec says that if result is null it's deemed that textDocument/rename is not possible at the current position (for example, if the user cursor is on a keyword).